### PR TITLE
Update contribution guide and pull request template

### PR DESCRIPTION
**Pull request by pedjak**
[Tekton project](https://github.com/tektoncd/community/blob/main/standards.md)
already has instructions that align with our values, and hence the important sections
are borrowed from there.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+# Submitter Checklist
+
+As the author of this PR, please check off the items in this checklist:
+
+- [ ] [Docs](https://github.com/redhat-developer/service-binding-operator/blob/master/CONTRIBUTING.md#docs) 
+  included if any changes are user facing
+- [ ] [Tests](https://github.com/redhat-developer/service-binding-operator/blob/master/CONTRIBUTING.md#tests)
+  included if any functionality added or changed. For bugfixes please include tests that can catch regressions
+- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
```
Comment (sadlerap) [inline .github/pull_request_template.md:23]: If we're going to copy-paste the commit message standard from tekton, we might as well link to our own copy directly: 
```suggestion
- [ ] Follows the [commit message standard](https://github.com/redhat-developer/service-binding-operator/blob/master/CONTRIBUTING.md#commits)
```

**Review by sadlerap — APPROVED** [review]
Besides one minor issue, this should be good.
/lgtm

File: .github/pull_request_template.md (hunk 1 above)
Comment (pedjak) [reply .github/pull_request_template.md]: done.

**Pull request merged**